app/controllers/auth_controller.rb

```diff
@@ -20,8 +20,16 @@ def create
     when :google
       user = GoogleAuthService.find_user_by_token(data_attrs[:google_id_token], data_attrs[:create_user])
       AuthToken.new(user: user)
-    when :okta, :facebook
-      user = "::#{method.to_s.capitalize}AuthService".constantize.find_user_by_token(data_attrs[:"#{method}_access_token"], data_attrs[:create_user])
+    when :okta
+      user = OktaAuthService.find_user_by_token(data_attrs[:okta_access_token], data_attrs[:create_user])
+      AuthToken.new(user: user)
+    when :facebook
+      # Handle both access tokens and ID tokens for Facebook
+      if data_attrs[:facebook_access_token]
```

Comment:
This appears to be the old method you're replacing. Shouldn't a check for the new method (line 30) come first?

Reply:
I'm indifferent as to the order of these, we should only be providing/validating one anyways.

I feel if a client passes multiple auth credentials it should do one of these things:
- attempt to validate one, and ignore the others. The one it chooses to validate should be considered undefined behavior.
- return an error about providing multiple credentials

Reply:
also, `facebook_access_token` isn't an old method currently, it is still the auth mechanism for Facebook login on the Android App. 

We had to make the change for iOS because Apple/Facebook were forcing an additional user confirmation prompt when using a facebook_access_token in the iOS app. If a similar problem shows up in Android we can switch, but there are no intentions of switching the Android mechanism at this time.

Reply:
It's not a big deal, but I feel that placing the `facebook_id_token` check first would indicate it's the preferred, modern method.